extensions/ql-vscode/src/remote-queries/shared/variant-analysis.ts

```diff
@@ -1,5 +1,92 @@
+import { Repository } from './repository';
+
+export interface VariantAnalysisSubmission {
```

Comment:
What will be the difference between the `VariantAnalysisSubmission` and the `VariantAnalysis`? It seems like they are very similar and might not need separate models?

Reply:
I'm also unclear about the difference between these two. I think the rest of the types introduced in this PR are clear enough, but these two could use comments explaining their purpose, or the names need to be much more descriptive.

About their actual purpose, is one the data we send to the API to submit the variant analysis and the other one is the response? They don't match up exactly since some fields like `databases` aren't in the response, so it's not an exact copy of an API response but rather we've added extra data into it (which is fine and good). So in that case I that reduces the benefit from having two separate types.

I suppose one benefit of having separate types is you can make more fields non-nullable, so that way just all the rest of the code can be simpler with fewer type checks.

Reply:
`VariantAnalysisSubmission` is a model that will be used within the extension "core" and I don't expect it to be used in the web views at all. The idea is that it contains all the information needed to request a variant analysis. It might be worth removing `VariantAnalysisSubmission` for now and introducing it along with the relevant functionality though. I think I'll do that!